src/channel.ts

```diff
@@ -1279,6 +1279,15 @@ export class Channel<StreamChatGenerics extends ExtendableGenerics = DefaultGene
           const ownMessage = event.user?.id === this.getClient().user?.id;
           const isThreadMessage = event.message.parent_id && !event.message.show_in_channel;
 
+          if (channel.data?.hidden) {
```

Comment:
Shouldn't we be rather calling `channel.show()` to be in sync with the back-end? Also the documentation mentions this method to show the channel again.

![image](https://user-images.githubusercontent.com/32706194/232702761-0c8dd096-4990-402e-be72-e74949a8df99.png)

Reply:
I tried that, but this would need significant refactoring on the client side since the method is async and returns a promise, and most of the code here is sync. Also, we didn't handle `channel.visible` before; that was a big issue too.

Reply:
My wild guess is the backend calls the `channel.hide()` API internally when a new message is received for a hidden channel.

Reply:
@khushal87 you think the backend calls `channel.hide()` on already hidden channel when a new message arrives?

Reply:
Not this exactly, but they locally update the hidden entity(if they have it in the DB) when the new message is received or execute the controller of the hidden logic.

Reply:
I tested the following workflow without the changes included in this PR:

1. A hides channel Boulder
2. B sends message to Boulder
3. Channel Boulder appears in the A's channel list again

@khushal87 Is this what you are trying to achieve by this PR?

https://ctrlv.tv/C0fw

Reply:
Yes @MartinCupela

Reply:
Lets say there are two channel list one with filter hidden true and the other with hidden false. If boulder was in channel list with filter hidden false and was hidden it shall move to channel with filter hidden true and once message.new is triggered, the channel should move back to channel list with hidden false.

Reply:
That is exactly what is happening on the video

Reply:
Oh sorry, I couldn't play the video for some reason. So do you think this PR can solve the issue?

Reply:
@khushal87 the video demo is not using your fix. So my concern is, whether this fix is actually needed.

Reply:
Can you show me the `message.new` event listener in the React SDK? Maybe you handle this case explicitly. Even we do it in RN SDK, and adding this change removes [the following lines](https://github.com/GetStream/stream-chat-react-native/blob/develop/package/src/components/ChannelList/hooks/listeners/useNewMessage.ts#L48).

Reply:
As discussed, I think instead of dispatching this event (which is wrong since its a backend event) ... we should just switch the hidden property to false here.

Reply:
Cool, I have removed the dispatch, but it will be great for backend to emit the `channel.visible` event, which I think is not emitted at all currently.

Reply:
@khushal87 I have pushed the change that  instead of dispatching the `channel.visible` event just switches the hidden property to false. As @vishalnarkhede proposed.